chsdi/views/search.py

```diff
@@ -393,11 +394,34 @@ def _add_feature_queries(self, queryText, timeFilter):
                     raise exc.HTTPBadRequest('Parameter seachLang is not supported for %s' % index)
                 self.sphinx.AddQuery(queryText, index=str(index))
 
+    def _box2d_transform(self, res):
+        """Reproject a ST_BOX2 from EPSG:21781 to SRID"""
+        try:
+            box2d = res['geom_st_box2d']
+            box_str = box2d[4:-1]
+            b = map(float, re.split(' |,', box_str))
+            shape = box(*b)
+            bbox = transform_shape(shape, self.DEFAULT_SRID, self.srid).bounds
+            res['geom_st_box2d'] = "BOX({} {},{} {})".format(*bbox)
+        except Exception:
+            raise exc.HTTPInternalServerError('Error while converting BOX2D to EPSG:{}'.format(self.srid))
+        return res
+
     def _parse_locations(self, res):
+
         if not self.returnGeometry:
             attrs2Del = ['x', 'y', 'lon', 'lat', 'geom_st_box2d']
             popAtrrs = lambda x: res.pop(x) if x in res else x
             map(popAtrrs, attrs2Del)
+        elif int(self.srid) not in (21781, 2056):
+            self._box2d_transform(res)
+            try:
+                pnt = (res['y'], res['x'])
+                x, y = transform_coordinate(pnt, self.DEFAULT_SRID, self.srid)
```

Comment:
in this case, if we are in `4326` we should use directly what is coming from sphinx as `lat` and `lon` (if exist)

prio 2